src/misc/JsonConfig.cpp

```diff
@@ -0,0 +1,187 @@
+#include "JsonConfig.h"
+#include <filesystem>
+#include <fstream>
+#include <spdlog/spdlog.h>
+#include <Utils/StringHelper.h>
+
+namespace fs = std::filesystem;
+
+JsonConfig::JsonConfig(std::string path) : mPath(path) {
+    // check if path exists and that it's a file
+    if (!fs::exists(path) || !fs::is_regular_file(path)) {
+        mIsNewConfig = true;
+        mJson = json::object();
+        return;
+    }
+
+    try {
+        std::ifstream ifs(path);
+        mJson = json::parse(ifs);
+    } catch (json::parse_error& e) {
+        SPDLOG_ERROR("Failed to parse JSON file: {}", e.what());
+
+        // If failure to parse, we will create a new file
+        mIsNewConfig = true;
+        mJson = json::object();
+    }
+}
+
+void JsonConfig::SetArbitraryType(std::string key, json::value_type value) {
+    auto keyParts = StringHelper::Split(key, ".");
+
+    if (keyParts.size() > 1) {
+        // nested key
+        json* currentJson = &mJson;
+        for (size_t i = 0; i < keyParts.size() - 1; i++) {
+            if (!currentJson->contains(keyParts[i])) {
+                // create new object
+                (*currentJson)[keyParts[i]] = json::object();
+            }
+            currentJson = &(*currentJson)[keyParts[i]];
+        }
+        (*currentJson)[keyParts[keyParts.size() - 1]] = value;
+    } else {
+        // not nested
+        mJson[key] = value;
+    }
+}
+
+json::value_type JsonConfig::GetArbitraryType(std::string key) {
+    auto keyParts = StringHelper::Split(key, ".");
+
+    // find the deepest nested key if it exists
+    json* currentJson = &mJson;
+    for (size_t i = 0; i < keyParts.size() - 1; i++) {
+        if (!currentJson->contains(keyParts[i])) {
+            return nullptr;
+        }
+        currentJson = &(*currentJson)[keyParts[i]];
+    }
+
+    // extract the value if possible
+    if (currentJson->contains(keyParts[keyParts.size() - 1])) {
+        return (*currentJson)[keyParts[keyParts.size() - 1]];
+    } else {
+        return nullptr;
+    }
+}
+
+// MARK: - Public API
+
+bool JsonConfig::IsNewConfig() {
+    return mIsNewConfig;
+}
+
+void JsonConfig::DeleteEntry(std::string key) {
+    auto keyParts = StringHelper::Split(key, ".");
+
+    // clear nested key, if after deletion parent is empty, delete it too
+    if (keyParts.size() > 1) {
+        json* currentJson = &mJson;
+        for (size_t i = 0; i < keyParts.size() - 1; i++) {
```

Comment:
could we just use
```cpp
for (auto keyPart : keyParts) {
   [...]
}
```
or 
```cpp
for (auto& keyPart : keyParts) {
   [...]
}
```
instead since `StringHelper::Split` gets us a `vector`?